### PR TITLE
General chopper simulation design

### PR DESCRIPTION
**Pull request by MichaelWedel**
This is for discussion of the design for the chopper simulation.

### CODE REVIEW

**Comment by MichaelWedel** [comment]
@DMSC-Instrument-Data/instrument-control I've started this kind of "design document" for the chopper. Please feel free to comment, add, modify!

File: docs/simulation-design.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+The `Bearing` and `Motor` classes will have some internal state machines according to the document we had before. Although they seem connected at first I think there's merit in separating them from each other (especially since the state machine of a `MechanicalBearing` will be very different from the `MagneticBearing`-machine). Furthermore it will be easier to test and change the state machines inside these classes compared to a monolithic one.
+
+The `Motor`, `Disc` and `ReferencePulseSource` will be coupled somehow, maybe this coupling happens in the actual `Chopper`-class?
+
+## Some technology choices
+
+** State machine library **
+I really like `Fysom` and unless it turns out to have some fundamental shortcoming for our use cases that can't be overcome I'd like to stick with it. It seems to do what we need and finding other modules would require investing some more time.
+
+** Observer pattern **
```
Comment (OwenArnold) [inline docs/simulation-design.md:24]: Functions are first-class citizens in Python, so call backs should be as easy as passing the function as an argument and invoking it on the "observed" object.

**Comment by OwenArnold** [comment]
@MichaelWedel This document looks to be somewhat behind now. I think documenting how the simulation works might still be a good idea though. If I where looking at plankton for the first time i'd what to know how the simulator worked.

**Comment by MikeHart85** [comment]
This PR/branch seems quite stale at this point. I propose we close it without merging.

#82 already covered providing general documentation on how Plankton works (although that will need to be updated once work on [the communication redesign](https://github.com/DMSC-Instrument-Data/documents/blob/master/design/plankton/CommunicationLayerDesign.md) is complete).

Are you happy with closing this @MichaelWedel?

**Comment by MichaelWedel** [comment]
Yes, this has been sitting here for ages, closing it now.

**Pull request closed without merging**